api/modules/movie/movie.helper.js

```diff
@@ -59,4 +75,27 @@ const constructSorting = (req) => {
   return { sortingField, sortingOrder };
 };
 
-module.exports = { constructPagination, constructsearchquery, constructFilterQueryObj, constructSorting };
+/**
+ * Function to construct year filteration for graph data
+ * @param {Object} req
+ * @return {Object}
+ */
+const constructYearFilteration = (req) => {
+  const minYear = req?.query?.minYear;
+  const maxYear = req?.query?.maxYear;
+
+  if (!minYear || !maxYear) {
+    return { $match: {} };
+  }
+
+  return {
+    $match: {
+      releaseDate: {
+        $gte: new Date(minYear, 0, 1), // Start of startYear
+        $lte: new Date(maxYear, 11, 31), // End of endYear
```

Comment:
why you are hardcoding the dates and month, you can use moment to get startOf and endOf month and years